scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -210,6 +211,9 @@ trait TypedPipe[+T] extends Serializable {
   /** prints the current pipe to stdout */
   def debug: TypedPipe[T] = onRawSingle(_.debug)
 
+  /** adds a description to the pipe */
+  def withDescription(description: String): TypedPipe[T] = new WithDescriptionTypedPipe[T](this, description)
```

Comment:
Sounds fine to me. Just wanted to see your thoughts here. I don't really see us writing a method like:

```scala
def describe[T <: WithDescription[T]](t: T, s: String): T = t.withDescription(s)
```
which is the only think I can think of us doing with this.

So, let's leave it as is.